library/src/main/java/com/schibsted/spain/barista/BaristaScrollActions.kt

```diff
@@ -0,0 +1,58 @@
+package com.schibsted.spain.barista
+
+import android.support.annotation.IdRes
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.PerformException
+import android.support.test.espresso.matcher.ViewMatchers.withId
+import android.support.test.espresso.matcher.ViewMatchers.withText
+import android.view.View
+import com.schibsted.spain.barista.custom.NestedEnabledScrollToAction.nestedScrollToAction
+import com.schibsted.spain.barista.internal.failurehandler.SpyFailureHandler
+import com.schibsted.spain.barista.internal.failurehandler.description
+import org.hamcrest.Matcher
+
+/**
+ * Scrolls in Espresso are not as great as we could except. For that reason, we will try
+ * to scroll several times instead of just one.
+ */
+object BaristaScrollActions {
+
+    // This value has been mathematically calculated and proven to be precisely the exact number of retries needed to always work.
+    // Not really, we just tried hundreds of times with different values and this seems to be the best one.
+    private val MAX_SCROLL_ATTEMPTS = 50
+
+    @JvmStatic
+    fun scrollTo(@IdRes id: Int) {
+        scrollWithMultipleAttempts(withId(id), failAtEnd = true)
+    }
+
+    @JvmStatic
+    fun scrollTo(text: String) {
+        scrollWithMultipleAttempts(withText(text), failAtEnd = true)
+    }
+
+    @JvmStatic
+    fun safelyScrollTo(@IdRes id: Int) {
+        scrollWithMultipleAttempts(withId(id), failAtEnd = false)
+    }
+
+    @JvmStatic
+    fun safelyScrollTo(text: String) {
+        scrollWithMultipleAttempts(withText(text), failAtEnd = false)
+    }
+
+    private fun scrollWithMultipleAttempts(matcher: Matcher<View>, failAtEnd: Boolean) {
+        val spyFailureHandler = SpyFailureHandler()
+        for (i in 0..MAX_SCROLL_ATTEMPTS) {
```

Comment:
Probably not too important, but this executes 51 times instead of 50, so not completely accurate :)
`1 .. MAX_SCROLL_ATTEMPTS` would fit better or `0 until MAX_SCROLL_ATTEMPTS`. But with the until you would have to change the i == MAX_SCROLL_ATTEMPTS

Reply:
Hum good point, I didn't think of that. The `i == MAX_SCROLL_ATTEMPTS` should have raised a warning in my head but it didn't 😅 

I'll go for the `1 .. MAX_SCROLL_ATTEMPTS`, it looks cleaner to me.

Thanks!!